### PR TITLE
SQS -> Pulsar topic routing enhancement

### PR DESCRIPTION
**Pull request by seburke**
Added functionality for routing to many topics based on a SQS message attribute

### CODE REVIEW

File: src/main/java/org/apache/pulsar/ecosystem/io/sqs/SQSRecord.java [hunk 1]
```diff
@@ -97,6 +97,10 @@ public void fail() {
 
     @Override
     public Optional<String> getDestinationTopic() {
-        return Optional.of(destination);
+        if (msg.getMessageAttributes().containsKey(SQSUtils.PULSAR_TOPIC_ATTRIBUTE)) {
+            return Optional.of(msg.getMessageAttributes().get(SQSUtils.PULSAR_TOPIC_ATTRIBUTE).getStringValue());
```
Comment (freeznet) [inline src/main/java/org/apache/pulsar/ecosystem/io/sqs/SQSRecord.java:101]: Can we have some validation with the topic name?

**Comment by Anonymitaet** [comment]
@freeznet thanks for your info. Now, @Huanli-Meng is the owner of connectors, she can help handle connector docs.

**Comment by seburke** [comment]
Just circling back with this if there is anything else needed to get this merged. 
Thanks!

**Review by freeznet — APPROVED** [review]
LGTM

**Comment by freeznet** [comment]
@nlu90 could you please help to review this PR when you have time, thanks.

**Review by nlu90 — CHANGES_REQUESTED** [review]
I'm a little concerned introducing the google guava and apache commons dependency for implementing one method.

Can we do it with the native java language features for now and avoid introducing such dependencies?

**Comment by seburke** [comment]
> I'm a little concerned introducing the google guava and apache commons dependency for implementing one method.
> 
> Can we do it with the native java language features for now and avoid introducing such dependencies?

I removed the new dependencies as you mentioned. Getting rid of the checkArgument method was straightforward. The isNotBlank method is a little more complex and seems unnecessary to implement when the awssdk has the same method and the implementations are pretty much the same so I switched back to that import since it is already imported anyway. If you want something else, happy to go that route, this just seemed easiest and I was trying to copy what the full pulsar topic validation is as closely as possible.

**Review by nlu90 — APPROVED** [review]
LGTM

**Comment by nlu90** [comment]
> > I'm a little concerned introducing the google guava and apache commons dependency for implementing one method.
> > Can we do it with the native java language features for now and avoid introducing such dependencies?
> 
> I removed the new dependencies as you mentioned. Getting rid of the checkArgument method was straightforward. The isNotBlank method is a little more complex and seems unnecessary to implement when the awssdk has the same method and the implementations are pretty much the same so I switched back to that import since it is already imported anyway. If you want something else, happy to go that route, this just seemed easiest and I was trying to copy what the full pulsar topic validation is as closely as possible.

Thanks for your contribution!

**Comment by freeznet** [comment]
@seburke would you please rebase the master so that the integration tests CI can be fixed, thanks.

**Pull request merged**